ibm/service/iampolicy/resource_ibm_iam_service_policy.go

```diff
@@ -453,17 +454,16 @@ func resourceIBMIAMServicePolicyCreate(d *schema.ResourceData, meta interface{})
 	}
 	if err != nil {
 		if v, ok := d.GetOk("iam_service_id"); ok && v != nil {
```

Comment:
instead of this condiiton 

shd we check if d.ID() doesn't start with "iam-" then add ?

Reply:
I guess, we may not have `d.ID()` while creating a policy..